src/util/modules.py

```diff
@@ -177,6 +177,11 @@ def uninstall_module(cr, module):
         if model == "ir.ui.view":
             for _, res_id in group:
                 remove_view(cr, view_id=res_id, silent=True)
+        elif model == "res.groups":
+            group_ids = [it[1] for it in group]
+            if group_ids:
+                cr.execute("DELETE FROM rule_group_rel WHERE group_id in %s", group_ids)
```

Comment:
Don't we risk here transforming a group-restricted rule into a _global_ rule?

I think this may be fine since we are removing the module owner of the group, and hopefully of the rules... It may be safer if we can verify that the module is indeed owner of all rules linked to this group.

Is this a common case or something specific to a module?

Reply:
You're right. What we need to do instead is to remove the rules before the groups.
We already have the issues with the server actions, hence the special case.

Actually, we could determine the order of deletion with a dependency graph based on the `ON DELETE` constraints. But I don't mind hardcoding model names in the `ORDER BY` clause of the query.

Reply:
Only if the rule is linked to only one group.

Reply:
That why removing the rules first will keep the `on delete restrict` in application.

Reply:
> What we need to do instead is to remove the rules before the groups.

If rules and groups are all from the removed module we are good. But in the case `module1.rule_x` being linked to `module2.group1,module2.group2` and we remove `module2`, then we should "do something" for `rule_x` of `module1`. Maybe disable it?

Reply:
In this case, we should just explicitly remove the rule and let it be recreated by the ORM.

Reply:
We just get the first case. I don't mind handling them with specific scripts until it become recurrent.